ssl/statem/extensions_clnt.c

```diff
@@ -108,8 +108,8 @@ static int use_ecc(SSL *s)
     return i < end;
 }
 
-int tls_construct_ctos_ec_pt_formats(SSL *s, WPACKET *pkt, X509 *x,
-                                     size_t chainidx, int *al)
+int tls_construct_ctos_ec_pt_formats(SSL *s, WPACKET *pkt, unsigned int context,
```

Comment:
Almost all functions will need at least 3 of those params. Hiding stuff like the SSL object and the WPACKET just doesn't seem helpful to me. I really do not see this as a readability gain - the opposite in fact...we will litter the code with lots of mystruct->pkt expressions or similar (or if not we have to have code to unpack the struct as well as pack it)